src/Infiltration/formulas/game.ts

```diff
@@ -5,35 +5,41 @@ export const MaxDifficultyForInfiltration = 3.5;
 // This value is typically denoted "lambda," and is the instantaneous rate of decay.
 // I.e. infils decay towards 0 at 0.1%/sec. This is a half-life of ~11.5 minutes.
 const DecayRate = -1e-6;
+// Tuning factor for the market demand multiplier.
+// Math.sqrt(1/TuningMult) infiltrations in a short window will reduce the rewards to near zero for a time.
+const TuningMult = 1.5e-2;
+const date = globalThis.Date;
 
 export const InfiltrationStateDefault = {
   lastChangeTimestamp: 0,
   infils: 0,
 };
-// Tracks an exponential moving average of number of successful infiltrations performed,
-// which decays back to 0. This state is only updated after a successful infil.
+// Tracks an exponential moving average of number of successful infiltrations performed, which decays back to 0.
 export const InfiltrationState = { ...InfiltrationStateDefault };
 
-function calculateCurrentInfils(timestamp: number): number {
-  return InfiltrationState.infils * Math.exp(DecayRate * (timestamp - InfiltrationState.lastChangeTimestamp));
+function calculateCurrentInfils(timestamp = date.now()): number {
+  // The clamp ensures that the market rate is always recovering, and not "stuck" at zero for a long time due to being effectively negative.
+  const clampedInfils = clampNumber(InfiltrationState.infils, 0.01, Math.sqrt(1 / TuningMult) + 0.07);
+  InfiltrationState.infils = clampedInfils * Math.exp(DecayRate * (timestamp - InfiltrationState.lastChangeTimestamp));
+  InfiltrationState.lastChangeTimestamp = timestamp;
+  return InfiltrationState.infils;
 }
 
 // Calculates the infiltration reward multiplier based on how many and how recent other infiltrations were completed.
 // Each infiltration completed reduces the demand for corporate espionage data for a little while, thus affecting the
 // market demand.
-export function calculateMarketDemandMultiplier(timestamp: number): number {
-  const infils = calculateCurrentInfils(timestamp);
+export function calculateMarketDemandMultiplier(): number {
```

Comment:
I didn't plumb timestamp through all these calls just for the fun of it. XD

If you don't pass in timestamp, you don't have a consistent value to make calculations with. This leads to small but subtle bugs where you are relying on multiple calls to Date.now() in succession to return the same value, which is *usually* but not *always* true.

Also, once the player starts the game, IMO the reward should be frozen; what you get at the end should be the same as what it said when you pressed "start". This is only possible with a consistent timestamp. You can still live-update the values in Intro before that point with a refresh hook, if you want.

Reply:
I reverted most of my changes, except for some minor ones to have live-updating on the intro screen. Rewards are frozen on start now.